### ISSUE TITLE
Tests finish without running all commands if Error is not a constructor

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#21428]
### Current behavior

I have noticed this rare but unfortunate behavior, same was noticed by other people on my team.
Sometimes a test passes, but when you look at it, it is missing lots of commands, as if it exited early. 
For example the test below should have a lot more commands, yet it has finished already somehow

<img width="845" alt="err1" src="https://user-images.githubusercontent.com/2212006/167695885-90111c2c-f905-47dc-a745-50cc0b060b1c.png">

When poking around, I noticed the Devtools is showing an error that is NOT reflected in the Command Log "ErrorConstructor is not a constructor" Seems somehow the `specWidow.Error` can be undefined, which causes a problem when we grab the user stack

<img width="602" alt="err2" src="https://user-images.githubusercontent.com/2212006/167696220-8df54c74-3cf1-4e2b-a579-6888b40b536f.png">

The error can be thrown only inside

```js
const captureUserInvocationStack = (ErrorConstructor, userInvocationStack) => {
  if (!userInvocationStack) {
    const newErr = new ErrorConstructor('userInvocationStack');
    userInvocationStack = newErr.stack; // if browser natively supports Error.captureStackTrace, use it (chrome) (must be bound)
    // otherwise use our polyfill on top.Error

    const captureStackTrace = ErrorConstructor.captureStackTrace ? ErrorConstructor.captureStackTrace.bind(ErrorConstructor) : Error.captureStackTrace;
    captureStackTrace(newErr, captureUserInvocationStack); // On Chrome 99+, captureStackTrace strips away the whole stack,
    // leaving nothing beyond the error message. If we get back a single line
    // (just the error message with no stack trace), then use the original value
    // instead of the trimmed one.

    if (newErr.stack.match('\n')) {
      userInvocationStack = newErr.stack;
    }
  }

  userInvocationStack = normalizedUserInvocationStack(userInvocationStack);
  return userInvocationStack;
};
```

Which receives the `ErrorConstructor` from its callers, but does not check if it is defined or a function before calling `new ...`

A few more places from the stack that call this function
<img width="1070" alt="err3" src="https://user-images.githubusercontent.com/2212006/167696488-1b69d3a5-c862-4a23-b68a-d37dc17de6c3.png">
<img width="1790" alt="err4" src="https://user-images.githubusercontent.com/2212006/167696492-619038a5-40a3-4ff8-a310-3d44d5d5486a.png">

A conditional breakpoint worked to stop there on undefined `Error` argument
<img width="756" alt="err5" src="https://user-images.githubusercontent.com/2212006/167696596-5a6dce05-b9ee-4a19-a7c0-27e5d1754375.png">

<img width="1536" alt="err7" src="https://user-images.githubusercontent.com/2212006/167696620-866f386f-e0d7-47bb-b866-8ad4ee4d99c4.png">

### Desired behavior

This is very rare but happens and is very concerning. Especially since it lowers the confidence in the passing tests. 
What I would like to do is a simple guard: if there is no `ErrorConstructor`, do not try to use it. Having no stack is preferred to skipping the rest of the test with PASS status.

### Test code to reproduce

Really don't know, since this function is called a lot at the start of the test. 
I will try to trace it better, but a little defensive programming would not hurt still

### Cypress Version

I have seen it in v8 and v9

**Comment by BlueWinds** [comment]
Hm, yeah, I see what you mean. I have only vague memories of this area of code, which I believe I was the last to work on, and I'm not certain how we'd get there with ErrorConstructor undefined. Would be very interested if you can figure that out.

Even if you can't track it down, I'm amenable to putting some "this shouldn't happen" guardrails around getting the stack trace - it's been a pain-point in a couple of different tickets, and "provide no stack trace" is certainly a better failure-state than "pass the test without executing the remaining commands." Happy to approve a PR in that direction, if you can't track down a better solution.

**Comment by bahmutov** [comment]
Funny that right above it in another function there is such guard

```js
const getInvocationDetails = (specWindow, config) => {
  if (specWindow.Error) {
    let stack = new specWindow.Error().stack; // note: specWindow.Cypress can be undefined or null
    // if the user quickly reloads the tests multiple times
    // firefox throws a different stack than chromium
    // which includes stackframes from cypress_runner.js.
    // So we drop the lines until we get to the spec stackframe (includes __cypress/tests)

    if (specWindow.Cypress && specWindow.Cypress.isBrowser('firefox')) {
      stack = stackWithLinesDroppedFromMarker(stack, '__cypress/tests', true);
    }

    const details = getSourceDetailsForFirstLine(stack, config('projectRoot')) || {};
    details.stack = stack;
    return details;
  }

  return;
};
```

**Comment by bahmutov** [comment]
More info. I added printing messages from different places when the `specWindow.Error` is undefined since we call to get the user invocation stack from a few places. Turns out, the error happens when we wrap `cy` commands like `task` and others.

<img width="846" alt="Screen Shot 2022-05-12 at 14 17 23" src="https://user-images.githubusercontent.com/2212006/168154645-d2f96393-ed1e-4025-a29c-8466167735cb.png">

<img width="582" alt="Screen Shot 2022-05-12 at 14 16 51" src="https://user-images.githubusercontent.com/2212006/168154672-2f2f9659-cf9a-423c-9f8a-9f29a12d4255.png">

<img width="602" alt="Screen Shot 2022-05-12 at 15 20 44" src="https://user-images.githubusercontent.com/2212006/168154705-2451764a-703f-4c8b-a0b9-db2ba45dab30.png">
